plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/transport/simulate/SimulateDocumentResult.java

```diff
@@ -0,0 +1,159 @@
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.plugin.ingest.transport.simulate;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentBuilderString;
+import org.elasticsearch.ingest.Data;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class SimulateDocumentResult implements Streamable, ToXContent {
+
+    private Data data;
+    private List<SimulateProcessorResult> processorResultList;
+    private Throwable failure;
+
+    public SimulateDocumentResult() {
+
+    }
+
+    public SimulateDocumentResult(Data data) {
+        this.data = data;
+    }
+
+    public SimulateDocumentResult(List<SimulateProcessorResult> processorResultList) {
+        this.processorResultList = processorResultList;
+    }
+
+    public SimulateDocumentResult(Throwable failure) {
+        this.failure = failure;
+    }
+
+    public boolean isFailed() {
+        if (failure != null) {
+            return true;
+        }
+        return false;
+    }
+
+    public boolean isVerbose() {
+        return this.processorResultList != null;
+    }
+
+    public Data getData() {
+        return data;
+    }
+
+    public List<SimulateProcessorResult> getProcessorResultList() {
+        return processorResultList;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        boolean isFailed = in.readBoolean();
+        boolean isVerbose = in.readBoolean();
+        if (isFailed) {
+            this.failure = in.readThrowable();
+        } else if (isVerbose) {
+            int size = in.readVInt();
+            processorResultList = new ArrayList<>();
+            for (int i = 0; i < size; i++) {
+                SimulateProcessorResult processorResult = new SimulateProcessorResult();
+                processorResult.readFrom(in);
+                processorResultList.add(processorResult);
+            }
+        } else {
+            String index = in.readString();
+            String type = in.readString();
+            String id = in.readString();
```

Comment:
yea looks good, that is what I meant. Do you like this solution? lemme know if you have other ideas!